rocAL/rocAL/source/pipeline/tensor.cpp

```diff
@@ -0,0 +1,374 @@
+        {
+            return sizeof(vx_int16);
+        }
+        break;
+        case RocalTensorDataType::UINT8:
+        {
+            return sizeof(vx_uint8);
+        }
+        case RocalTensorDataType::UINT32:
+        {
+            return sizeof(vx_uint32);
+        }
+        case RocalTensorDataType::INT32:
+        {
+            return sizeof(vx_int32);
+        }
+        break;
+        default:
+            throw std::runtime_error("tensor data_type not valid");
+    }
+}
+
+bool operator==(const rocALTensorInfo &rhs, const rocALTensorInfo &lhs)
+{
+    return (rhs.dims() == lhs.dims() &&
+            rhs.mem_type() == lhs.mem_type() &&
+            rhs.data_type() == lhs.data_type() &&
+            rhs.color_format() == lhs.color_format() &&
+            rhs.layout() == lhs.layout());
+}
+
+void rocALTensorInfo::reallocate_tensor_roi_buffers()
+{
+    _roi = std::make_shared<std::vector<RocalROI>>(_batch_size);
+
+    if(_roi->size() > 0)
+        _roi->clear();
+    _roi->resize(_batch_size);
+    for (unsigned i = 0; i < _batch_size; i++)
+    {
+        _roi->at(i).x1 = 0;
+        _roi->at(i).y1 = 0;
+        _roi->at(i).x2 = _max_dims.at(0);
+        _roi->at(i).y2 = _max_dims.at(1);
+    }
+}
+
+rocALTensorInfo::rocALTensorInfo() : _type(Type::UNKNOWN),
+                                _num_of_dims(0),
+                                _dims({}),
+                                _mem_type(RocalMemType::HOST),
+                                _data_type(RocalTensorDataType::FP32){}
+
+rocALTensorInfo::rocALTensorInfo(
+    std::vector<unsigned> dims,
+    RocalMemType mem_type,
+    RocalTensorDataType data_type) : _type(Type::UNKNOWN),
+                                _dims(dims),
+                                _mem_type(mem_type),
+                                _data_type(data_type)
+{
+    _batch_size = dims.at(0);
+    _num_of_dims = dims.size();
+    _data_size = tensor_data_size(data_type);
+    for(unsigned i = 0; i < _num_of_dims; i++)
+        _data_size *= dims.at(i);
+
+    if(_num_of_dims <= 3)
+        _is_image = false;
+}
+
+void rocALTensor::update_tensor_roi(const std::vector<uint32_t> &width, const std::vector<uint32_t> &height)
+{
+    if(_info.is_image())
+    {
+        auto max_dims = _info.max_dims();
+        unsigned max_width = max_dims.at(0);
+        unsigned max_height = max_dims.at(1);
+
+        if (width.size() != height.size())
+            THROW("Batch size of Tensor height and width info does not match")
+
+        if (width.size() != info().batch_size())
+            THROW("The batch size of actual Tensor height and width different from Tensor batch size " + TOSTR(width.size()) + " != " + TOSTR(info().batch_size()))
+
+        for (unsigned i = 0; i < info().batch_size(); i++)
+        {
+            if (width[i] > max_width)
+            {
+                ERR("Given ROI width is larger than buffer width for tensor[" + TOSTR(i) + "] " + TOSTR(width[i]) + " > " + TOSTR(max_width))
```

Comment:
Should this be an err? If you want the lib to exit then keep it as err.
if you are assigning the width to max_width then it should be a warning.